source/indexes.txt

```diff
@@ -24,12 +24,146 @@ Optimize Queries by Using Indexes
+   :language: scala
+   :emphasize-lines: 18-21
+
+Single Field Index
+------------------
+
+The following example creates an ascending index on the specified field:
+
+.. literalinclude:: /includes/indexes/single-field.scala
+   :language: scala
+   :start-after: start-index-single
+   :end-before: end-index-single
+   :dedent:
+
+To learn more about single field indexes, see the :ref:`scala-single-field-index`
+guide.
+
+Compound Index
+--------------
+
+The following example creates a compound indexes on fields named ``runtime``
+and ``year``.
+
+.. literalinclude:: /includes/indexes/compound-field.scala
+   :language: scala
+   :start-after: start-index-compound
+   :end-before: end-index-compound
+   :dedent:
+
+To learn more about compound indexes, see the :ref:`scala-compound-index` guide.
+
+Geospatial Index
+----------------
+
+The following example creates a 2dsphere index on the specified field that
+contains GeoJSON objects:
+
+.. literalinclude:: /includes/indexes/single-field.scala
+   :language: scala
+   :start-after: start-create-geospatial-index
+   :end-before: end-create-geospatial-index
+   :dedent:
+
+For more information on 2dsphere indexes, see the 
+:manual:`2dsphere Indexes </core/indexes/index-types/geospatial/2dsphere/>`
+guide in the {+mdb-server+} manual.
+
+For more information about the GeoJSON type, see the 
+:manual:`GeoJSON Objects </reference/geojson/>` guide in
+the {+mdb-server+} manual.
+
+Atlas Search Index Management
+-----------------------------
+
+The following sections contain code examples that describe how to manage 
+Atlas Search indexes.
+
+To learn more about search indexes, see the :ref:`scala-atlas-search-index` guide.
+
+Create Search Index
+~~~~~~~~~~~~~~~~~~~
+
+The following example creates an Atlas Search index on the specified field:
+
+.. literalinclude:: /includes/indexes/atlas-search-index.scala
+   :language: scala
+   :start-after: start-create-search-index
+   :end-before: end-create-search-index
+   :dedent:
+
+List Search Indexes
+~~~~~~~~~~~~~~~~~~~
+
+The following example prints a list of Atlas Search indexes in the specified 
+collection:
+
+.. literalinclude:: /includes/indexes/atlas-search-index.scala
+   :language: scala
+   :start-after: start-list-search-indexes
+   :end-before: end-list-search-indexes
+   :dedent:
+
+Update Search Indexes
+~~~~~~~~~~~~~~~~~~~~~
+
+The following example updates an existing Atlas Search index with the specified
+new index definition:
+
+.. literalinclude:: /includes/indexes/atlas-search-index.scala
+   :language: scala
+   :start-after: start-update-search-indexes
+   :end-before: end-update-search-indexes
+   :dedent:
+
+Delete Search Indexes
+~~~~~~~~~~~~~~~~~~~~~
+
+The following example deletes an Atlas Search index with the specified name:
+
+.. literalinclude:: /includes/indexes/atlas-search-index.scala
+   :language: scala
+   :start-after: start-drop-search-index
+   :end-before: end-drop-search-index
+   :dedent:
+
+Drop Index
```

Comment:
S: maybe move this section before Atlas Search? It seems a little out of place after it